### DISCUSSION TITLE
Is there any developing guideline for prometheus_client?

### DISCUSSION THREAD

**Discussion by ElapsedSoul** [prometheus/client_golang#941] [General]
The document [doc](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#pkg-types) of guide we have now just tell what but not why.
The demo just let us to use but can't let us know why, I can just change it a little but can't write one for my idea easily.
Is there any chart for module design let us know the theory of the design of client library framework?
like scrapy do:
https://docs.scrapy.org/en/latest/topics/architecture.html
So that ,we can know the function for every module of client library code, we can know what can we do in each code module.

Is anyone could help?

**Comment by kakkoyun** [comment]
We don't have something like that as far as I know. We could convert this to an issue and describe what you need. I would be happy to help.
And sorry for the late response.
